tests/contract_testcases/substrate/structs/parse_structs_10.dot

```diff
@@ -12,6 +12,12 @@ strict digraph "tests/contract_testcases/substrate/structs/parse_structs_10.sol"
 	note_13 [label="recursive field 'c'\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:9:17-20"]
 	diagnostic_14 [label="struct 'B' has infinite size\nlevel Error\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:12:20-21"]
 	note_15 [label="recursive field 'd'\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:13:17-20"]
+	diagnostic_16 [label="struct 'A' has infinite size\nlevel Error\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:16:20-21"]
+	note_17 [label="recursive field 'd'\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:17:17-20"]
+	diagnostic_18 [label="struct 'A' has infinite size\nlevel Error\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:16:20-21"]
+	note_19 [label="recursive field 'b'\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:18:17-20"]
+	diagnostic_20 [label="struct 'A' has infinite size\nlevel Error\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:16:20-21"]
+	note_21 [label="recursive field 'c'\ntests/contract_testcases/substrate/structs/parse_structs_10.sol:19:17-20"]
```

Comment:
We shouldn't be raising repeated errors.

Reply:
I think you can add multiple notes to the same error.

Reply:
They are different errors. There are multiple erroneous fields on struct A so we expect multiple errors. The details are not shown here. But we still expect them.

Reply:
The error message is exactly the same for all of them. You should have only one error message with multiple notes.

Reply:
Your function `flag_as_infinite` is adding multiple times the same error message to Diagnostics. Instead of doing that, you can create a single diagnostic for `struct 'A' has infinite size` with multiple notes. `struct Diagnostic` has a `notes: Vec<Note>` field. The problem is not the test.

Reply:
`solc` doesn't even report the offending struct fields, nor does it report about all recursive structs (if there are multiple). But I changed it anyways, looks like this now:

```bash
error: struct 'A' has infinite size
   ┌─ /home/glow/code/solang/tests/contract_testcases/substrate/structs/parse_structs_10.sol:16:20
   │
16 │             struct A {
   │                    ^
17 │                 D d;
   │                 --- recursive field 'd'
18 │                 B b;
   │                 --- recursive field 'b'
19 │                 C c;
   │                 --- recursive field 'c'
```